File: examples/quickstart-typescript/app/fixtures.ts
/**
 * Sample data used by contract tests.
 *
 * Import in tests/contract/stele_context.ts to wire real domain objects into
 * the Stele fixture.
 */

import type { Item, Order, User } from "./domain.js";

export function sampleUser(): User {
  return { id: "usr-001", email: "alice@example.com", status: "active" };
}

export function sampleOrders(): Order[] {
  return [
    {
      id: "ord-001",
      total: 49.99,
      items: [
        { sku: "WIDGET-A1", price: 29.99, qty: 1 },
        { sku: "WIDGET-B2", price: 20.0, qty: 1 },
      ],
    },
    {
      id: "ord-002",
      total: 14.5,
      items: [{ sku: "GADGET-C3", price: 14.5, qty: 1 }],
    },
  ];
}

/** Flat item list used by the SKU_FORMAT checker. */
export function sampleItems(): Item[] {
  return sampleOrders().flatMap((o) => o.items);
}

File: examples/quickstart-python/app/fixtures.py
"""
Sample data used by contract tests.

Import these in conftest.py to wire real domain objects into the Stele fixture.
"""

from __future__ import annotations

from decimal import Decimal

from .domain import Item, Order, User


def sample_user() -> User:
    return User(id="usr-001", email="alice@example.com", status="active")


def sample_orders() -> list[Order]:
    ord1 = Order(
        id="ord-001",
        total=Decimal("49.99"),
        items=[
            Item(sku="WIDGET-A1", price=Decimal("29.99"), qty=1),
            Item(sku="WIDGET-B2", price=Decimal("20.00"), qty=1),
        ],
    )
    ord2 = Order(
        id="ord-002",
        total=Decimal("14.50"),
        items=[
            Item(sku="GADGET-C3", price=Decimal("14.50"), qty=1),
        ],
    )
    return [ord1, ord2]


def sample_items() -> list[Item]:
    """Flat item list used by the SKU_FORMAT checker."""
    orders = sample_orders()
    return [item for order in orders for item in order.items]
